test/clustertest/tests/MultipleLeaderSyncTest.cpp

```diff
@@ -0,0 +1,91 @@
+#include "../BedrockClusterTester.h"
+
+struct MultipleLeaderSyncTest : tpunit::TestFixture {
+    MultipleLeaderSyncTest()
+        : tpunit::TestFixture("MultipleLeaderSyncTest",
+                              BEFORE_CLASS(MultipleLeaderSyncTest::setup),
+                              AFTER_CLASS(MultipleLeaderSyncTest::teardown),
+                              TEST(MultipleLeaderSyncTest::test)
+                             ) { }
+
+    BedrockClusterTester* tester;
+
+    void setup() {
+        // create a 5 node cluster
+        tester = new BedrockClusterTester(BedrockClusterTester::FIVE_NODE_CLUSTER, {"CREATE TABLE test (id INTEGER NOT NULL PRIMARY KEY, value TEXT NOT NULL)"}, _threadID);
+
+        // make sure the whole cluster is up
+        ASSERT_TRUE(tester->getBedrockTester(0)->waitForStates({"LEADING", "MASTERING"}));
+        for (int i = 1; i <= 4; i++) {
+            ASSERT_TRUE(tester->getBedrockTester(i)->waitForStates({"FOLLOWING", "SLAVING"}));
+        }
+
+        // shut down primary leader
+        tester->stopNode(0);
+
+        // Wait for node 1 to be leader.
+        ASSERT_TRUE(tester->getBedrockTester(1)->waitForStates({"LEADING", "MASTERING"}));
+
+        // increase the delta between primary and secondary leader
+        runTrivialWrites(5000, 4);
+
+        // shut down secondary leader
+        tester->stopNode(1);
+
+        // Wait for node 2 to be leader.
+        ASSERT_TRUE(tester->getBedrockTester(2)->waitForStates({"LEADING", "MASTERING"}));
+
+        // give secondary leader a few commits to sync
```

Comment:
Nitpicky, but this comment, if required, seems like it should be more or less the same as the one above for an identical line.